libr/asm/arch/8051/8051_ass.c

```diff
@@ -1232,36 +1247,38 @@ static bool mnem_swap(char const*const*arg, ut16 pc, ut8**out) {
 }
 
 static bool mnem_xrl(char const*const*arg, ut16 pc, ut8**out) {
-	if (!r_str_casecmp (arg[0], "a")) {
-		if (is_indirect_reg (arg[1])) {
-			return singlearg_register (0x66, arg[1], out);
+	ut16 dest_hexadecimal;
+	if (parse_hexadecimal (arg[0], &dest_hexadecimal)) {
+		if (!r_str_casecmp (arg[1], "a")) {
+			return singlearg_direct (0x62, arg[0], out);
+		}
+		if (arg[1][0] == '#') {
+			(*out)[0] = 0x63;
```

Comment:
out[0] = 0x63; // if you change the function signature to ut8 *out instead of **out